python_modules/dagster/dagster/_core/ext/client.py

```diff
@@ -21,7 +23,7 @@ def run(
         *,
         context: OpExecutionContext,
         extras: Optional[ExtExtras] = None,
-    ) -> None: ...
+    ) -> Union["MaterializeResult", Tuple["MaterializeResult", ...]]: ...
```

Comment:
Would very much want this type signature to not be "dual state" like this. We shouldn't specialize the one versus many use case

Reply:
Yeah this is really unfortunate and I tried to avoid this at first. It has to do with the fact that the internal machinery complains if you return a tuple with a single `MaterializeResult` when expecting a single result, and I wanted to be able to just  return `ExtClient.run` from an `@asset`.

Possible solutions:

1. Go with the iterator approach described above
2. Adjust framework to accept a 1-tuple when expecting a single result
3. Always return a tuple and require the user to unpack to a single object for a 1-tuple
4. Provide a separate method `get_materialize_result` that returns a single `MaterializeResult`.

Atm I like solution (1).